### PR TITLE
[rush-lib] Fix an issue where rush update can not handle dependenciesMeta in pnpm 

### LINKED ISSUE #4452: [rush] rush update can not handle dependenciesMeta in pnpm
## Summary
The PNPM has a setting called dependenciesMeta, see more details on [PNPM](https://pnpm.io/package_json#dependenciesmeta)
It is used to set up the injected settings for packages. 

Right now, when users set up this in `package.json`, `rush update` does not update the `pnpm-lock.yaml`

## Repro steps

This is a bug I found when I working on [pnpm-sync-prototype](https://github.com/g-chao/pnpm-sync-prototype)
So, we can use this repo to reproduce the bug.

1. Change the injected settings for any workspace dependencies. In this example, let's modify `package.json` for `apps/app3` , add `"dependenciesMeta": {
    "lib1": {
      "injected": true
    }
  }`
3. Run `rush update`
4. Check `pnpm-lock.yaml` under `common/config/rush`, you will see there is **no change** for `pnpm-lock.yaml` 
5. However, if you check `common/temp` folder, you will see `pnpm-lock.yaml` and `pnpm-lock-preinstall.yaml` **are different**
6. Now, delete the `common/temp` folder, and run `rush install`, you will the see PNPM complains `ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with ../../apps/app3/package.json`

The expected behavior is that after we make some changes to `dependenciesMeta` in `package.json`, `rush update` should update the `pnpm-lock.yaml` under `common/config/rush`. So that, the `rush install` will not fail later. 

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | 5.109.1 |
| `rushVersion` from rush.json? | 5.109.1 |
| `useWorkspaces` from rush.json? | pnpm workspace |
| Operating system? | Mac |
| Would you consider contributing a PR? | Yes |
| Node.js version (`node -v`)? | v16.20.2 |

### PR DESCRIPTION
**Pull request by g-chao**
## Summary
This PR fixes #4452

## Details

I added some additional logic for `shrinkwrapIsUpToDate`

If `dependenciesMeta` settings are different between `package.json` and `pnpm-lock.yaml`, then `shrinkwrapIsUpToDate` return false.

## How it was tested

Manually tested with a sample rush repo.

### CODE REVIEW

File: libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts [hunk 1]
```diff
@@ -235,6 +247,54 @@ export class WorkspaceInstallManager extends BaseInstallManager {
...
+
+      for (const rushProject of this.rushConfiguration.projects) {
+        const packageJson: PackageJsonEditor = rushProject.packageJsonEditor;
+        const packageJsonFilePath: string = packageJson.filePath;
+
+        // get the relativePackagePath for a package, to align with the value in pnpm-lock.yaml
+        const relativePackagePath: string = path.relative(
+          commonTempFolder,
+          packageJsonFilePath.replace('/package.json', '')
+        );
```
Comment (iclanton) [inline libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts:266]: You shouldn't need to call `path.relative` for each project, given that the relative path from the `common/temp` folder to the repo root is the same for all projects. You should be able to calculate the relative path between `commonTempFolder` and the repo root and then concatenate that with the project folder relative path.

File: common/changes/@microsoft/rush/chao-support-injected-settings_2023-12-13-18-48.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Fix an issue where rush update can not handle dependenciesMeta changes in pnpm",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/chao-support-injected-settings_2023-12-13-18-48.json:5]:
```suggestion
      "comment": "Fix an issue where Rush does not detect changes to the `dependenciesMeta` field in project's `package.json` files, so may incorrectly skip updating/installation.",
```

File: common/changes/@rushstack/node-core-library/chao-support-injected-settings_2023-12-13-18-48.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@rushstack/node-core-library",
+      "comment": "Add dependenciesMeta interface in packageJson",
```
Comment (iclanton) [inline common/changes/@rushstack/node-core-library/chao-support-injected-settings_2023-12-13-18-48.json:5]:
```suggestion
      "comment": "Add the `dependenciesMeta` property to the `INodePackageJson` interface.",
```

File: common/changes/@rushstack/node-core-library/chao-support-injected-settings_2023-12-13-18-48.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+      "type": "patch"
```
Comment (iclanton) [inline common/changes/@rushstack/node-core-library/chao-support-injected-settings_2023-12-13-18-48.json:6]:
```suggestion
      "type": "minor"
```

**Review by octogonz — APPROVED** [review]

**Pull request merged**